sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json

```diff
@@ -814,44 +877,46 @@
                     "condition": "[equals(parameters('kind'), 'featurestore')]",
                     "type": "connections",
                     "apiVersion": "2022-05-01",
-                    "name": "[variables('offlineStoreConnectionName')]",
+                    "name": "[if(empty(parameters('offline_store_connection_name')), 'empty', parameters('offline_store_connection_name'))]",
```

Comment:
will this create a ws connection with name "empty", or some otherr logic will prevent this connection to be created?

Reply:
or why the offline_store_connetion_name will be empty?

Reply:
no, when connection name is empty, the deployment condition will never be satisfied. I put it "empty" because arm template validation still checks that connection name to make sure it's a non-empty string otherwise it will throw even the deployment step condition will never be met and the deployment step will not be executed.

Reply:
which condition prevents this "when connection name is empty, the deployment condition will never be satisfied" ?

Reply:
is that the "dependsOn" condition prevents this?

Reply:
no, sdk will always give a valid offline store connection name, after we fix the logic. But in anyway, we need to a non empty string as name in that deployment step.